tests/unit/src/test/scala/tests/markdown/BaseMarkdownSuite.scala

```diff
@@ -80,6 +80,31 @@ abstract class BaseMarkdownSuite extends tests.BaseSuite {
     }
   }
 
+  def checkMulti(
+      name: TestOptions,
+      original: Map[String, String],
+      settings: Settings = baseSettings,
+      onOutput: String => Unit = _ => ()
+  ): Unit = {
+    test(name) {
+      val reporter = newReporter()
+      val context = newContext(settings, reporter)
+      original.foreach { case (filename, contents) =>
```

Comment:
Looks like the mdoc task sorts files by name (`val files = IO.inputFiles(settings).sorted`), so since file order appears to be important for repro, try sorting `original` by file name